read.js

```diff
@@ -0,0 +1,95 @@
+const { MongoClient } = require('mongodb');
+
+async function main() {
+    /**
+     * Connection URI. Update <username>, <password>, and <your-cluster-url> to reflect your cluster.
+     * See http://bit.ly/NodeDocs_lauren for more details
+     */
+    const uri = "mongodb+srv://<username>:<password>@<your-cluster-url>/test?retryWrites=true&w=majority";
+
+    /**
+     * The Mongo Client you will use to interact with your database
+     * See bit.ly/Node_MongoClient for more details
+     */
+    const client = new MongoClient(uri);
+
+    try {
+        // Connect to the MongoDB cluster
+        await client.connect();
+
+        // Make the appropriate DB calls
+
+        // Find the listing named "Infinite Views" that we created in create.js
+        await findListingByName(client, "Infinite Views");
+
+        // Find up to 5 listings with at least 4 bedrooms and at least 2 bathrooms
+        // If you recently ran create.js, a listing named Beautiful Beach House should be included in the results 
+        await findListingsWithMinimumBedroomsBathroomsAndMostRecentReviews(client, 4, 2, 5);
```

Comment:
This works fine for the example, but it is approaching a readability issue. This could go either way, as there isn't a preset rule to this, but I would probably rewrite this to use destructuring:

```js
async function findListingsWithMinimumBedroomsBathroomsAndMostRecentReviews(client, { 
  minimumNumberOfBedrooms=0,
  minimumNumberOfBathrooms=0,
  maximumNumberOfResults=Number.MAX_SAFE_INTEGER
}={}) {
```

Then, when the function is invoked, it would be called by doing:

```js
await findListingsWithMinimumBedroomsBathroomsAndMostRecentReviews(client, {
  minimumNumberOfBedrooms: 4,
  minimumNumberOfBathrooms: 2,
  maximumNumberOfResults: 5
});
```

Pros:
- Makes it clear what parameters you are passing in
- Allows for flexibility in passing parameters (you don't need to worry about order)
- Easier to make parameters optional / have default values

Cons:
- More Verbose
- If you **don't** want to have any parameters be optional, it makes it a little harder to enforce that

See [this stackoverflow question](https://stackoverflow.com/questions/12826977/multiple-arguments-vs-options-object) for more discussion on the subject.

Reply:
I would also move to passing the collection rather than the client.

Reply:
Thanks for the tips on destructuring @daprahamian!  I've updated the code.  Is it ok/normal to have some functions with desctructuring and some without?  

@codepope In the original version of the code, I passed the collection.  I had to switch to passing the client when it was decided that the first blog post had to list the databases instead of running a query.  To prevent having to re-explain the main() function in every post, I've been passing the client.